### ISSUE TITLE
Typescript Support for extending native classes

### ISSUE THREAD

**Issue by georgeedwards** [NativeScript/NativeScript#2287]
### Tell us about the problem

The [docs](https://docs.nativescript.org/runtimes/android/generator/extend-class-interface.html) that discuss extending android classes only demonstrate JS. The same code in Typescript throws an error:

```
var constructorCalled = false;
var MyButton = android.widget.Button.extend({
        //constructor
        init: function() {
...
```

>[ts] Property 'extend' does not exist on type 'typeof IntentService'.

### Which platform(s) does your issue occur on?
Android

NB: there is another issue [here](https://github.com/NativeScript/docs/issues/376) on the docs repo discussing issues with Typescript and extending classes.

**Comment by vjoao** [comment]
To make `tsc` happy you need to download the TypeScript definitions for Android and iOS.

**Comment by georgeedwards** [comment]
@vjoao I have done `npm i tns-platform-declarations --save-dev`, and referenced the downloaded files. I am successfully referencing many of the platform APIs defined in these files, yet I am still getting tsc errors about extending classes?

**Comment by tsonevn** [comment]
merged with #2295
